### PR TITLE
Seed names

### LINKED ISSUE #2173: [bug] Accepted character range for seed names
The server communicates that only [`-A-Za-z0-9`] are valid characters for seed names, but this replay shows that it was possible to use a `#` character in the name. And it causes some wonky stuff when looking at other scores, for example.

https://hanab.live/replay/507894

### PR DESCRIPTION
**Pull request by DrAnax**
Closes #2173

### CODE REVIEW

**Comment by DrAnax** [comment]
Optionally:

Fix existing seeds (search seeds containing `#`, `_`, `;`):

1. Test
  `SELECT id, seed, REGEXP_REPLACE(seed, '[#_;]', '-', 'g') AS becomes FROM "games" WHERE seed ~ '[#_;]';`

2. Change
  `UPDATE "games" SET seed = REGEXP_REPLACE(seed, '[#_;]', '-', 'g') WHERE seed ~ '[#_;]';`

3. Confirm
  `SELECT * FROM "games" WHERE seed ~ '[#_;]';`

File: server/src/misc.go [hunk 1]
```diff
@@ -306,16 +306,13 @@ func toSnakeCase(str string) string {
...
-func fixTableName(name string) string {
+func truncateTrimCheckEmpty(name string) string {
 	// Truncate long table names
 	// (we do this first to prevent wasting CPU cycles on validating extremely long table names)
 	if len(name) > MaxGameNameLength {
 		name = name[0 : MaxGameNameLength-1]
 	}
 
-	// Remove any non-printable characters, if any
-	name = removeNonPrintableCharacters(name)
```
Comment (Zamiell) [inline server/src/misc.go:317]: why

Comment (DrAnax) [reply server/src/misc.go:317]: removeNonPrintableCharacters was obsolete; valid table/seed name is handled by regex

File: server/src/command_replay_create.go [hunk 2]
```diff
@@ -45,7 +45,12 @@ func commandReplayCreate(ctx context.Context, s *Session, d *CommandData) {
 	} else if d.Source == "json" {
 		// Before creating a new game and emulating the actions,
 		// ensure that the submitted JSON does not have any obvious errors
-		if valid, _ := isJSONValid(s, d); !valid {
+
+		if d.GameJSON == nil {
```
Comment (Zamiell) [inline server/src/command_replay_create.go:49]: this should be part of the isJSONValid function

Comment (DrAnax) [reply server/src/command_replay_create.go:49]: it will break command_table_create.go:L91:96

Comment (Zamiell) [reply server/src/command_replay_create.go:49]: oh I see

**Comment by Zamiell** [comment]
great work, thanks

**Pull request merged**